src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/DetectPLINQNops.Fixer.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.NetCore.Analyzers.Usage
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = DetectPLINQNopsAnalyzer.RuleId), Shared]
+    public sealed class DetectPLINQNopsFixer : CodeFixProvider
+    {
+        private static readonly string[] removableEnds = new string[] { "ToList", "ToArray", "AsParallel" };
```

Comment:
What about `ToDictionary` and others?

Reply:
Smart, totally forgot about them. ToHashSet has an effect for the enumeration, as can ToDictionary. ToHashSet could reduce the number of loops, ToDictionary can act as a select for the key and could not be discarded. In that case we would want to remove just AsParallel right?

Reply:
I would say so.